File: Frontend/src/app/_models/equivalence-request.ts
import { GUID } from '../../utils/guid';
import { ExemptedCourse } from './exempted-course';
import { Approval } from './approval';

export interface EquivalenceRequest {
  id: GUID;
  studentId: string;
  firstName?: string;
  lastName?: string;
  hostUniversityName?: string;
  hostCourseName: string;
  hostCourseCode: string;
  hostCourseEcts: number;
  fileName: string;
  exemptedCourse: ExemptedCourse;
  instructorApproval?: Approval;
  additionalNotes: string;
  isCanceled: boolean;
  isApproved: boolean;
  isRejected: boolean;
  isArchived: boolean;
}

File: Backend/Entities/EquivalenceRequest.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Threading.Tasks;

namespace Backend.Entities
{
    /// <summary>A class for representing an course equivalence request.</summary>
    public class EquivalenceRequest : Form
    {
        [Key]
        public Guid Id { get; set; }
        [Required]
        public string StudentId { get; set; }
        public string HostCourseName { get; set; }
        public string HostCourseCode { get; set; }
        public double HostCourseECTS { get; set; }
        public DateTime SubmissionDate { get; set; } = DateTime.Now;
        public string FileName { get; set; }
        public byte[] Syllabus { get; set; }
        public ExemptedCourse ExemptedCourse { get; set; }
        public Approval InstructorApproval { get; set; }
        public string AdditionalNotes { get; set; } = "";
        public bool IsCanceled { get; set; } = false;
        public bool IsRejected { get; set; } = false;
        public bool IsApproved { get; set; } = false;
        public bool IsArchived { get; set; } = false;
    }
}
